src/librustc/ty/walk.rs

```diff
@@ -118,8 +118,12 @@ fn push_subtypes<'tcx>(stack: &mut TypeWalkerStack<'tcx>, parent_ty: Ty<'tcx>) {
             stack.extend(substs.substs.types().rev());
         }
         ty::TyGenerator(_, ref substs, ref interior) => {
-            stack.extend(substs.substs.types().rev());
             stack.push(interior.witness);
+            stack.extend(substs.substs.types().rev());
+        }
+        ty::TyGeneratorWitness(ts) => {
+            // FIXME: Should we skip the binder here?
+            stack.extend(ts.skip_binder().iter().cloned().rev());
```

Comment:
`TyFnPtr` skips binders blindly in the next 5 lines, so sure.

Reply:
Ok. Removed this FIXME.